src/StreamJsonRpc.Tests/MessagePackFormatter.cs

```diff
@@ -46,12 +60,25 @@ public MessagePackFormatter(bool compress)
             this.compress = compress;
         }
 
+        /// <inheritdoc/>
+        public JsonRpc Rpc
+        {
+            private get => this.rpc;
+            set
+            {
+                Verify.Operation(this.rpc == null, "This formatter already belongs to another JsonRpc instance. Create a new instance of this formatter for each new JsonRpc instance.");
+
+                this.rpc = value;
+            }
+        }
+
         /// <inheritdoc/>
         public JsonRpcMessage Deserialize(ReadOnlySequence<byte> contentBuffer)
         {
-            return this.compress
-                ? (JsonRpcMessage)LZ4MessagePackSerializer.Typeless.Deserialize(contentBuffer.AsStream())
-                : (JsonRpcMessage)MessagePackSerializer.Typeless.Deserialize(contentBuffer.AsStream());
+
+            return this.compress ?
+                (JsonRpcMessage)MessagePackSerializer.Typeless.Deserialize(contentBuffer.AsStream(), MessagePackSerializerOptions.LZ4Default) :
+                (JsonRpcMessage)MessagePackSerializer.Typeless.Deserialize(contentBuffer.AsStream());
         }
```

Comment:
For each time we look at `compress` and do something different, let's replace that with just a `MessagePackSerializerOptions` field set based on the `compress` parameter passed in to our constructor. Then all the branching that appears elsewhere in the class can go away.